deploy/kubernetes/metricbeat/README.md

```diff
@@ -2,30 +2,145 @@
+- role: worker
+- role: worker
+- role: worker
+```
+
+After that, we can create a cluster using the command below.
+
+```
+$ kind create cluster --config=cluster-config.yaml
+```
+
+Notice that our cluster was created using the configuration we created above. The name of our cluster is the default one, `kind`. To interact with the cluster, you only need to specify the cluster name as a context in kubectl:
+
+```
+$ kubectl cluster-info --context kind-kind
+```
+
+
+#### Option 2. Run on GKE
+
+1. Create a GKE cluster. You can find information on how to do that [here](https://cloud.google.com/kubernetes-engine/docs/deploy-app-cluster).
+
+2. Connect to the newly created cluster. You can do that by running the following:
+```
+$ gcloud container clusters get-credentials <cluster name> --zone <zone> --project <project name>
+```
+
+
+#### Run Metricbeat on Kubernetes
+
+You can check the nodes of the newly created cluster by running the following:
+
+```
+$ kubectl get nodes
+```
+
+Metricbeat gets some metrics from [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics). You can download the project. You can run, for example:
+
+```
+$ kubectl apply -f kube-state-metrics/examples/standard
+```
+
+To deploy Metricbeat to Kubernetes, clone this repository. Don't forget to set the docker image version inside the [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml).
+Remember that you can find [Metricbeat Docker images here](https://www.docker.elastic.co/r/beats/metricbeat).
+After that, run:
+
+```
+$ kubectl apply -f beats/deploy/kubernetes/metricbeat
+```
+
+You can check the pods of your cluster in the namespace `kube-system`. For that, you just run:
+
+```
+$ kubectl get pods -o wide --namespace=kube-system
+```
+
+You should be able to see each pod's node.
+If you are using Linux/Unix/macOS you can request all pods whose names start with metricbeat:
+
+```
+$ kubectl get pods -o wide --namespace=kube-system | grep ^metricbeat
+```
+
+
+#### Visualizing data in Kibana
+
+To visualize your data in Kibana, we will use Elastic Cloud. For that do the following:
+1. [Log in](https://cloud.elastic.co/home) to your Elastic Cloud account.
+2. Select the target deployment.
+3. Navigate to the **Analytics** endpoint and select **Discover**. To see Metricbeat data, make sure the predefined `metricbeat-*` index pattern is selected.
+4. To load a Kibana dashboard, you have to select **Dashboard** on the side menu. For example, you can see data about your Metricbeat DaemonSet on the dashboard *[Metricbeat Kubernetes] DaemonSets*. If, by any chance, you don't have dashboards downloaded in Kibana, you can go [here](https://www.elastic.co/guide/en/beats/metricbeat/current/load-kibana-dashboards.html) to find a solution for that.
+
+If you can't see any data on the index, then you probably haven't set the YAML manifest file as you should. To do that, open the [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml) and update the following:
```

Comment:
Personally I would rephrase this bit.
If you are using Elastic cloud, you need first to deploy your Elastic cloud cluster to get the Id and username/password before you can deploy the elastic agent, you can't do the other way around.

So I suggest that instead of saying, "if you can't see any data ...", (since there is no other way that could work)... `Now deploy the elastic agent with these credentials from the newly created Elastic cloud...`